src/main/java/com/microsoft/sqlserver/jdbc/KeyVaultCredential.java

```diff
@@ -5,86 +5,118 @@
 
 package com.microsoft.sqlserver.jdbc;
 
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.Future;
-
-import com.microsoft.aad.adal4j.AuthenticationContext;
-import com.microsoft.aad.adal4j.AuthenticationResult;
-import com.microsoft.aad.adal4j.ClientCredential;
-import com.microsoft.azure.keyvault.authentication.KeyVaultCredentials;
+import com.azure.core.annotation.Immutable;
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.util.logging.ClientLogger;
+import com.microsoft.aad.msal4j.ClientCredentialFactory;
+import com.microsoft.aad.msal4j.ClientCredentialParameters;
+import com.microsoft.aad.msal4j.ConfidentialClientApplication;
+import com.microsoft.aad.msal4j.IAuthenticationResult;
+import com.microsoft.aad.msal4j.IClientCredential;
+import com.microsoft.aad.msal4j.SilentParameters;
+import java.net.MalformedURLException;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import reactor.core.publisher.Mono;
 
 
 /**
- * 
- * An implementation of ServiceClientCredentials that supports automatic bearer token refresh.
- *
+ * An AAD credential that acquires a token with a client secret for an AAD application.
  */
-class KeyVaultCredential extends KeyVaultCredentials {
-
-    SQLServerKeyVaultAuthenticationCallback authenticationCallback = null;
-    String clientId = null;
-    String clientKey = null;
-    String accessToken = null;
+@Immutable
+class KeyVaultCredential implements TokenCredential {
+    private final ClientLogger logger = new ClientLogger(KeyVaultCredential.class);
+    private final String clientId;
+    private final String clientSecret;
+    private String authorization;
+    private ConfidentialClientApplication confidentialClientApplication;
 
-    KeyVaultCredential(String clientId) throws SQLServerException {
+    /**
+     * Creates a KeyVaultCredential with the given identity client options.
+     *
+     * @param clientId
+     *        the client ID of the application
+     * @param clientSecret
+     *        the secret value of the AAD application.
+     */
+    KeyVaultCredential(String clientId, String clientSecret) {
+        Objects.requireNonNull(clientSecret, "'clientSecret' cannot be null.");
```

Comment:
@lilgreenbird 
This is not how we handle exception messages. Error messages need to be localized and also make sure to throw SQLServerException. Please make sure to change this in all occurrences.